pkg/cluster/delete.go

```diff
@@ -259,9 +259,21 @@ func (m *manager) deleteRoleDefinition(ctx context.Context) error {
 
 func (m *manager) Delete(ctx context.Context) error {
 	resourceGroup := stringutils.LastTokenByte(m.doc.OpenShiftCluster.Properties.ClusterProfile.ResourceGroupID, '/')
+	//In edge case of CRG not being managedBy ARO, we have a different delete path
+	//we will assume normal case and set rgManagedByARO to true CRG is managedby ARO
+	rgManagedByARO := true
+	rg, err := m.resourceGroups.Get(ctx, resourceGroup)
+	if err != nil {
+		m.log.Warnf("failed to get resourceGroup %s", err)
+	} else {
+		if rg.ManagedBy == nil || *rg.ManagedBy == "" || !strings.EqualFold(*rg.ManagedBy, m.doc.OpenShiftCluster.ID) {
+			rgManagedByARO = false
+			m.log.Infof("cluster resource group not managed by aro %s", *rg.Name)
+		}
+	}
```

Comment:
Does it make sense to move this and everything which is under `if rgManagedByARO` condition into a separate not exported method?

Reply:
And another nit: please add spaces after `//` in comments.